dist/step_definitions/elements.js

```diff
@@ -24,13 +24,7 @@ function _asyncToGenerator(fn) { return function () { var gen = fn.apply(this, a
 
 (0, _cucumber.defineSupportCode)(function ({ When, Then }) {
   When(/^I wait for "([^"]*)" of the "([^"]*)" element$/, function (condition, elementName) {
-    const timeout = parseInt(_config2.default.elementsVisibilityTimeout) * 1000;
-
-    if (this.currentPage[elementName] instanceof protractor.ElementArrayFinder) {
-      return browser.wait(protractor.ExpectedConditions[condition](this.currentPage[elementName].get(0)), timeout);
-    }
-
-    return browser.wait(protractor.ExpectedConditions[condition](this.currentPage[elementName]), timeout);
+    return condition === 'visibilityOf' ? this.currentPage.waitForVisibilityOf(elementName) : this.currentPage.waitForInvisibilityOf(elementName);
```

Comment:
I wouldn't do that like this, because you might as well wait for `elementToBeClickable`, and with this construct if condition is true you wait for visibility of Element, in any other case you are waiting for invisibilityOf